python/ShipGeoConfig.py

```
@@ -125,23 +147,23 @@ def load_from_root_file(root_file, key: str = "ShipGeo") -> Config:
-            config = Config()
-            config.loads_json(content_str)
+            # JSON format — parse to typed config
+            d = json.loads(content_str)
+            return ShipGeoConfig.from_dict(d)
         else:
-            # Assume pickle format - unpickle it
-            # Convert to bytes for pickle (using latin-1 encoding)
+            # Pickle format (legacy) — unpickle, convert to typed config
             pickle_bytes = content_str.encode("latin-1")
-            config = pickle.loads(pickle_bytes)
+            legacy = pickle.loads(pickle_bytes)  # nosec: intentional for legacy ROOT files
```

Ruff (0.15.9)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)